Storage/api.h

```diff
@@ -124,31 +127,29 @@ struct PerFieldRESTfulHandlerGenerator {
     const std::string restful_url_prefix = this->restful_url_prefix;
     const std::string field_name = input_field_name;
 
-    // Data handler.
     using entry_t = typename ENTRY_TYPE_WRAPPER::entry_t;
     using key_t = typename ENTRY_TYPE_WRAPPER::key_t;
-    using GETHandler = DataHandlerImpl<GET, specific_field_t, entry_t, key_t>;
-    using POSTHandler = DataHandlerImpl<POST, specific_field_t, entry_t, key_t>;
-    using PUTHandler = DataHandlerImpl<PUT, specific_field_t, entry_t, key_t>;
-    using DELETEHandler = DataHandlerImpl<DELETE, specific_field_t, entry_t, key_t>;
+    using top_level_operation_t = semantics::rest::operation::top_level_operation_for_field_t<specific_field_t>;
+
+    using GETHandler = DataHandlerImpl<GET, top_level_operation_t, specific_field_t, entry_t, key_t>;
+    using POSTHandler = DataHandlerImpl<POST, top_level_operation_t, specific_field_t, entry_t, key_t>;
+    using PUTHandler = DataHandlerImpl<PUT, top_level_operation_t, specific_field_t, entry_t, key_t>;
+    using DELETEHandler = DataHandlerImpl<DELETE, top_level_operation_t, specific_field_t, entry_t, key_t>;
 
     using SchemaRoutesGenerator = SchemaHandlerImpl<entry_t>;
```

Comment:
:+1: 21c07c2

Interestingly, I've moved it down, and then up again, in hopes to minimize the diff size of this CR. Fixed, and, while on this note, indented the large lambda above as well.